src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPResultSetCursorTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.tvp;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.Properties;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+@RunWith(JUnitPlatform.class)
+public class TVPResultSetCursorTest extends AbstractTest {
+
+    private static Connection conn = null;
+    static Statement stmt = null;
+
+    static BigDecimal[] expectedBigDecimals = {new BigDecimal("12345.12345"), new BigDecimal("125.123"), new BigDecimal("45.12345")};
+    static String[] expectedBigDecimalStrings = {"12345.12345", "125.12300", "45.12345"};
+
+    static String[] expectedStrings = {"hello", "world", "!!!"};
+
+    static Timestamp[] expectedTimestamps = {new Timestamp(1433338533461L), new Timestamp(14917485583999L), new Timestamp(1491123533000L)};
+    static String[] expectedTimestampStrings = {"2015-06-03 06:35:33.4610000", "2442-09-18 18:59:43.9990000", "2017-04-02 01:58:53.0000000"};
+
+    private static String tvpName = "TVPResultSetCursorTest_TVP";
+    private static String srcTable = "TVPResultSetCursorTest_SourceTable";
+    private static String desTable = "TVPResultSetCursorTest_DestinationTable";
+
+    /**
+     * Test a previous failure when using server cursor and using the same connection to create TVP and result set.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testServerCursors() throws SQLException {
+        conn = DriverManager.getConnection(connectionString);
```

Comment:
Can you add one test case for 3 result sets (2 TVPType and one Non-TVP)  & 2 / 3 statements for these resultsets?